tools/src/launcher/Main.hs

```diff
@@ -488,24 +501,21 @@ spawnNode (path, args, mbLogPath) = do
             createDirectoryIfMissing True (directory lp)
             (lp,) <$> openFile lp AppendMode
         Nothing -> do
-            tempdir <- fromString <$> getTemporaryDirectory
-            -- FIXME (adinapoli): `Shell` from `turtle` was giving us no-resource-leak guarantees
-            -- via the `Managed` monad, which is something we have lost here, and we are back to manual
-            -- resource control. In this case, however, shall we really want to nuke the file? It seems
-            -- something useful to have lying around in the filesystem. We should probably close the
-            -- `Handle`, though, but if this program is short lived it won't matter anyway.
-            IO.openTempFile tempdir "cardano-node-output.log"
+            let cr = createProcInherit path args
+            (_, stdO, stdE, pid) <- Process.createProcess cr
+            case stdO of
+                Just o -> do
+                    _ <- asyncLog o (fromJust stdE) pid NNode
+                    return (path, o)
+                Nothing -> do
+                    return (path, stdin)
```

Comment:
Sorry, but I don't understand anything now. You create a process here and then you create it again below. It doesn't make any sense to me.

Reply:
True this is bollocks. I changed everything.